webmention.php

```diff
@@ -81,13 +114,34 @@ public static function parse_query($wp) {
     if (!array_key_exists('webmention', $wp->query_vars)) {
       return;
     }
-
-    $content = file_get_contents('php://input');
-    parse_str($content);
+    else {
+      // check if the end point has expired
+      $valid_ticks = array( 0, -1, -2 );
```

Comment:
My suggestion would be:

* add a setting to enable/disable nonce support
* using `$nonce = wp_create_nonce( 'my-action_'.$post->ID );` so it is possible to use all other wp_nonce native features
* using the filter to return a "stateless" UID

What are you thinking?